src/miral/system_compositor_window_manager.cpp

```diff
@@ -0,0 +1,312 @@
+{
+    return false;
+}
+
+bool miral::SystemCompositorWindowManager::handle_pointer_event(MirPointerEvent const* /*event*/)
+{
+    return false;
+}
+
+int miral::SystemCompositorWindowManager::set_surface_attribute(
+    std::shared_ptr<ms::Session> const& /*session*/,
+    std::shared_ptr<ms::Surface> const& surface,
+    MirWindowAttrib attrib,
+    int value)
+{
+    return surface->configure(attrib, value);
+}
+
+void miral::SystemCompositorWindowManager::on_session_added(std::shared_ptr<mir::scene::Session> const& /*session*/) const
+{
+}
+
+void miral::SystemCompositorWindowManager::on_session_removed(std::shared_ptr<mir::scene::Session> const& session) const
+{
+    if (focus_controller->focused_session() == session)
+    {
+        auto const next_session = session_coordinator->successor_of({});
+        if (next_session)
+            focus_controller->set_focus_to(next_session, next_session->default_surface());
+        else
+            focus_controller->set_focus_to(next_session, {});
+    }
+}
+
+void miral::SystemCompositorWindowManager::on_session_ready(std::shared_ptr<mir::scene::Session> const& session) const
+{
+    focus_controller->set_focus_to(session, session->default_surface());
+}
+
+void miral::SystemCompositorWindowManager::handle_raise_surface(
+    std::shared_ptr<ms::Session> const& /*session*/,
+    std::shared_ptr<ms::Surface> const& /*surface*/,
+    uint64_t /*timestamp*/)
+{
+}
+
+void miral::SystemCompositorWindowManager::handle_request_move(
+    std::shared_ptr<ms::Session> const& /*session*/,
+    std::shared_ptr<ms::Surface> const& /*surface*/,
+    MirInputEvent const* /*event*/)
+{
+}
+
+void miral::SystemCompositorWindowManager::handle_request_resize(
+    std::shared_ptr<ms::Session> const& /*session*/,
+    std::shared_ptr<ms::Surface> const& /*surface*/,
+    MirInputEvent const* /*event*/,
+    MirResizeEdge /*edge*/)
+{
+}
+
+void miral::SystemCompositorWindowManager::advise_output_create(const miral::Output &/*output*/)
+{
+    // When a display gets added, we reposition all surfaces across their outputs, most likely to fit them to the screen.
+    for (auto const& surface_output_pair : output_map)
+    {
+        if (auto surface = surface_output_pair.first.lock())
+        {
+            auto const output_id = surface_output_pair.second;
+            reposition_surface_in_output(surface, output_id);
+        }
+    }
+}
+
+void miral::SystemCompositorWindowManager::advise_output_update(const miral::Output &/*updated*/, const miral::Output &/*original*/)
+{
+    std::lock_guard lock{mutex};
+
+    // When a display gets updated, we reposition all surfaces across their outputs, most likely to fit them to the screen.
+    for (auto const& surface_output_pair : output_map)
+    {
+        if (auto surface = surface_output_pair.first.lock())
+        {
+            auto const output_id = surface_output_pair.second;
+            reposition_surface_in_output(surface, output_id);
```

Comment:
@AlanGriffiths I made `reposition_surface_in_output` a separate function because I would expect us to only have to update the output that changed in this scenario instead of all the outputs. However, I am still updating all of the outputs in the current implementation. Would updating a single output be the correct thing to do?

Reply:
Yes, I think that would be a sane optimization